lib/torrent.js

```diff
@@ -600,12 +606,18 @@ class Torrent extends EventEmitter {
             this._debug('piece verified %s', index)
             this._markVerified(index)
           } else {
+            this.pieces[index] = new Piece(length || this.pieceLength)
+            this.bitfield.set(index, false)
+            this.select(index, index, 1)
+            this.files.forEach(file => {
+              if (file.done && file.includes(index)) file.done = false
+            })
```

Comment:
Maybe I'm missing something, but I don't see why this is necessary.

By default, if the pieces is not valid:
- The `this.pieces[index]` element is not emptied 
- The `this.bitfield` is already `false`
- The reselection of the piece is not needed because the `this.bitfield` was not changed
- The `file.done` is not changed because we still don't have the piece

Is this right @ftreesmilo ?

Reply:
In a rescan, a fully downloaded file with a corruption issue will think it has the chunk and later find out it doesn't.